x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/snapshotlifecycle/SnapshotLifecyclePolicyTests.java

```diff
@@ -139,7 +146,12 @@ public static SnapshotLifecyclePolicy randomSnapshotLifecyclePolicy(String id) {
             randomAlphaOfLength(4),
             randomSchedule(),
             randomAlphaOfLength(4),
-            config);
+            config,
+            randomRetention());
+    }
+
+    private static SnapshotRetentionConfiguration randomRetention() {
+        return new SnapshotRetentionConfiguration(TimeValue.parseTimeValue(randomTimeValue(), "random retention generation"));
```

Comment:
I think this should occasionally return an empty configuration - doing so would have caught the `equals` NPE bug I also commented on.

Reply:
:+1:

Reply:
Did you make this change? I don't see it.

Reply:
Hah, I did make it, and then I did the refactoring to move the `randomRetention` into a single place, which ended up removing it again, so I made this change again, thanks for catching it :)